controllers/managedfusionoffering_controller.go

```diff
@@ -192,8 +193,41 @@ func (r *ManagedFusionOfferingReconciler) own(resource metav1.Object) error {
 
 // All the below functions are placeholder for offering plugin integration
 
+type dfOfferingConfig struct {
+	usableCapacityInTiB     int
+	onboardingValidationKey string
+}
+
 // This function is a placeholder for offering plugin integration
 func pluginReconcile(r *ManagedFusionOfferingReconciler) (ctrl.Result, error) {
+	var validationErrorExist bool
+	var dfOfferingConfigData dfOfferingConfig
+
+	r.Log.Info(fmt.Sprintf("Validating %s offering CR config", r.managedFusionOffering.Spec.Kind))
+	if usableCapacityInTiBAsString, found := r.managedFusionOffering.Spec.Config["usableCapacityInTiB"]; !found {
+		r.Log.Error(fmt.Errorf("%s offering CR config does not have a usableCapacityInTiB entry", r.managedFusionOffering.Spec.Kind),
+			"missing usableCapacityInTiB info in offering CR config")
+		validationErrorExist = true
+	} else if usableCapacityInTiB, err := strconv.Atoi(usableCapacityInTiBAsString); err != nil {
+		r.Log.Error(err, fmt.Sprintf("usableCapacityInTiB in %s offering CR config does not have an integer value",
+			r.managedFusionOffering.Spec.Kind))
+		validationErrorExist = true
+	} else {
+		dfOfferingConfigData.usableCapacityInTiB = usableCapacityInTiB
+	}
+
+	if onboardingValidationKeyAsString, found := r.managedFusionOffering.Spec.Config["onboardingValidationKey"]; !found {
+		r.Log.Error(fmt.Errorf("%s offering CR config does not have a onboardingValidationKey entry", r.managedFusionOffering.Kind),
+			"missing onboardingValidationKey info in offering CR config")
+		validationErrorExist = true
+	} else {
+		dfOfferingConfigData.onboardingValidationKey = onboardingValidationKeyAsString
+	}
+
+	if validationErrorExist {
+		return ctrl.Result{}, fmt.Errorf("invalid %s offering CR config", r.managedFusionOffering.Spec.Kind)
+	}
+
 	return ctrl.Result{}, nil
```

Comment:
It makes sense to have the config parsing code as a separate function that returns the dataFoundationSpec structure 

```suggestion
	return dataFoundationSpec{
		usableCapacityInTib:  usableCapacityInTib,
		onboardingValidationKey: onboardingValidationKey,
	}
```

Reply:
Updated